<!DOCTYPE html>
<html dir="ltr" lang="en-gb">

<!-- Mirrored from forum.abinit.org/viewtopic.php?f=19&t=2219&view=print by HTTrack Website Copier/3.x [XR&CO'2014], Fri, 20 Sep 2024 22:02:15 GMT -->
<!-- Added by HTTrack --><meta http-equiv="content-type" content="text/html;charset=UTF-8" /><!-- /Added by HTTrack -->
<head>
<meta charset="utf-8" />
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="robots" content="noindex" />

<title>ABINIT Discussion Forums &bull; [SOLVED] how to run new tests?</title>

<link href="styles/flat-style/theme/print.css" rel="stylesheet">
<link href="styles/flat-style/theme/bidi.css" rel="stylesheet">
</head>
<body id="phpbb" class="ltr">
<div id="wrap" class="wrap">
	<a id="top" class="top-anchor" accesskey="t"></a>

	<div id="page-header">
		<h1>ABINIT Discussion Forums</h1>
		<p>The meeting place for ABINIT users and developers<br /><a href="index.html">https://forum.abinit.org/</a></p>

		<h2>[SOLVED] how to run new tests?</h2>
		<p><a href="viewtopic10a1.html?f=19&amp;t=2219">https://forum.abinit.org/viewtopic.php?f=19&amp;t=2219</a></p>
	</div>

	<div id="page-body" class="page-body">
		<div class="page-number">Page <strong>1</strong> of <strong>1</strong></div>
					<div class="post">
				<h3>[SOLVED] how to run new tests?</h3>
				<div class="date">Posted: <strong>Fri Jul 19, 2013 9:58 am</strong></div>
				<div class="author">by <strong>rangel</strong></div>
				<div class="content">Hi everyone, <br /><br />Now that the old tests script has been removed, I would like to use the new python script.<br /><br />I build abinit in a directory:<br />[ABINIT]/build<br /><br />Then, I compile and at the end I go to the directory tests, as usual:<br />cd [ABINIT]/build/tests<br /><br />Then I launch a test using the new python script (following Matteo's presentation in the ABINIT workshop):<br />[ABINIT]/tests/runtests.py v1[1] -v <br /><br />This is the output I get:<br /><span style="color: #400000"><br />Running on u-adidas -- system Linux -- ncpus 4 -- Python 2.7.2 -- runtests.py-0.4<br />mpi_nprocs 1, omp_nthreads 0, py_nthreads 1<br />('Initalizing JobRunner for sequential runs.',)<br />Loading database from file: /media/DISK/7.5.1-private/tests/test_suite.cpkl<br />Test_suite directory already exists! Old files will be removed<br />Running ntests = 1, MPI_nprocs = 1, py_nthreads = 1...<br />Command<br /> returned exit_code: 1<br /><br /><strong class="text-strong">[v1][t01] fldiff.pl fatal error:<br />Unable to open input file /media/DISK/7.5.1-private/build2/tests/Test_suite/v1_t01/t01.out<br /></strong><br /></span><br /><br />Indeed, t01.out does not exist.<br />However, if I go to /Test_suite/v1_t01 and there I launch the command :<br /> /media/DISK/7.5.1-private/build2/src/98_main/abinit &lt; /media/DISK/7.5.1-private/build2/tests/Test_suite/v1_t01/t01.stdin &gt; /media/DISK/7.5.1-private/build2/tests/Test_suite/v1_t01/t01.stdout 2&gt; /media/DISK/7.5.1-private/build2/tests/Test_suite/v1_t01/t01.stderr<br /><br />It works fine and it produces t01.out<br /><br />What am I doing wrong? How should I run the tests?<br /><br />Best wishes, <br />Tonatiuh Rangel</div>
			</div>
			<hr />
					<div class="post">
				<h3>Re: how to run new tests?</h3>
				<div class="date">Posted: <strong>Fri Jul 19, 2013 2:10 pm</strong></div>
				<div class="author">by <strong>gmatteo</strong></div>
				<div class="content">Hi Tonatiuh,<br /><br /><blockquote class="uncited"><div>[v1][t01] fldiff.pl fatal error:<br />Unable to open input file /media/DISK/7.5.1-private/build2/tests/Test_suite/v1_t01/t01.out<br /></div></blockquote><br /><br />There should be an error message in /media/DISK/7.5.1-private/build2/tests/Test_suite/v1_t01/t01.stderr that <br />should explain why the subprocess failed.<br /><br />Matteo</div>
			</div>
			<hr />
					<div class="post">
				<h3>Re: how to run new tests?</h3>
				<div class="date">Posted: <strong>Fri Jul 19, 2013 2:55 pm</strong></div>
				<div class="author">by <strong>rangel</strong></div>
				<div class="content">Hi Matteo, <br /><br />I see in stdout:<br />Commande timeout inaccessible. Essayer cea_timeout<br /><br />so there is something missing about timing<br />Do you know how to fix this?<br /><br />I already tried to give a value for timeout:<br />../..//tests/runtests.py v1[1] -v --timeout=1000<br />But it did not work.<br /><br /><br /><br />Best<br />Tonatiuh</div>
			</div>
			<hr />
					<div class="post">
				<h3>Re: how to run new tests?</h3>
				<div class="date">Posted: <strong>Fri Jul 19, 2013 3:23 pm</strong></div>
				<div class="author">by <strong>gmatteo</strong></div>
				<div class="content">By default, the script executes the tests under the control of the linux tool timeout (if found) <br />so that we kill the job if the run gets stuck. It seems that the version of timeout installed in your machine <br />is not working properly. To bypass this problem you can use the option -t [--timeout] that allows the user to define <br />the timeout value in seconds. <br />Setting --timeout to zero is equivalent to disabling the use of the timeout tool.<br />Try with:<br /><br />   runtests.py -t 0 [other_options]<br /><br />and let me know if it works.<br /><br />Cheers,<br />Matteo</div>
			</div>
			<hr />
					<div class="post">
				<h3>Re: how to run new tests?</h3>
				<div class="date">Posted: <strong>Fri Jul 19, 2013 4:19 pm</strong></div>
				<div class="author">by <strong>rangel</strong></div>
				<div class="content">Thanks,  now it works!</div>
			</div>
			<hr />
			</div>

	<div id="page-footer" class="page-footer">
		<div class="page-number">All times are <span title="Europe/Brussels">UTC+02:00</span><br />Page <strong>1</strong> of <strong>1</strong></div>
			<div class="copyright">
				<p>Powered by <a href="https://www.phpbb.com/">phpBB</a>&reg; Forum Software &copy; phpBB Limited
				</p>
							</div>
	</div>
</div>

</body>

<!-- Mirrored from forum.abinit.org/viewtopic.php?f=19&t=2219&view=print by HTTrack Website Copier/3.x [XR&CO'2014], Fri, 20 Sep 2024 22:02:15 GMT -->
</html>
